pkg/collect/sonobuoy_results.go

```diff
@@ -0,0 +1,164 @@
+)
+
+type CollectSonobuoyResults struct {
+	Collector    *troubleshootv1beta2.Sonobuoy
+	BundlePath   string
+	Namespace    string // this is not used
+	ClientConfig *rest.Config
+	Client       kubernetes.Interface
+	Context      context.Context
+	RBACErrors
+}
+
+func (c *CollectSonobuoyResults) Title() string {
+	return getCollectorName(c)
+}
+
+func (c *CollectSonobuoyResults) IsExcluded() (bool, error) {
+	return isExcluded(c.Collector.Exclude)
+}
+
+func (c *CollectSonobuoyResults) Collect(progressChan chan<- interface{}) (CollectorResult, error) {
+	namespace := DefaultSonobuoyNamespace
+	if c.Collector.Namespace != "" {
+		namespace = c.Collector.Namespace
+	}
+
+	podName := DefaultSonobuoyAggregatorPodName
+	resultsPath := DefaultSonobuoyAggregatorResultsPath
+	containerName := DefaultSonobuoyAggregatorContainerName
+
+	_, err := c.Client.CoreV1().Pods(namespace).Get(c.Context, podName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sonobuoy pod %s in namespace %s: %v", podName, namespace, err)
+	}
+
+	reader, ec, err := sonobuoyRetrieveResults(c.Context, c.Client, c.ClientConfig, namespace, podName, containerName, resultsPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve sonobuoy results: %v", err)
+	}
+
+	output := NewResult()
+
+	ec2 := make(chan error, 1)
+
+	go func() {
+		defer close(ec2)
+
+		gz, err := gzip.NewReader(reader)
+		if err != nil {
+			ec2 <- errors.Wrap(err, "failed to create gzip reader")
+			return
+		}
+		defer gz.Close()
+
+		tr := tar.NewReader(gz)
+
+		for {
+			header, err := tr.Next()
+			if err == io.EOF {
+				return
+			} else if err != nil {
+				ec2 <- errors.Wrap(err, "failed to read tar header")
+				return
+			}
+			if header.Typeflag != tar.TypeReg {
+				continue
+			}
+			filename := filepath.Clean(header.Name) // sanitize the filename
+			err = output.SaveResult(c.BundlePath, filepath.Join("sonobuoy", filename), tr)
+			if err != nil {
+				ec2 <- errors.Wrapf(err, "failed to save result for %s", filename)
+				return
+			}
+		}
+	}()
+
+	err = <-ec2
+	if err != nil {
+		_, _ = io.Copy(io.Discard, reader) // ensure the stream is closed
+		return nil, fmt.Errorf("failed to write sonobuoy results: %v", err)
+	}
+
+	err = <-ec
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve sonobuoy results: %v", err)
+	}
+
+	return output, nil
+}
+
+// sonobuoyRetrieveResults copies results from a sonobuoy run into a Reader in tar format.
+// It also returns a channel of errors, where any errors encountered when writing results
+// will be sent, and an error in the case where the config validation fails.
+func sonobuoyRetrieveResults(
+	ctx context.Context, client kubernetes.Interface, restConfig *rest.Config, namespace, podName, containerName, path string,
+) (io.Reader, <-chan error, error) {
+	ec := make(chan error, 1)
+
+	restClient := client.CoreV1().RESTClient()
+	req := restClient.Post().
```

Comment:
Nit: Could you log the command being run? When testing this I couldn't get results so wanted to test using `kubectl`. Grabbing the parameters from logs would be nice

Some high level logs like "results were collected", "could not collect anything... "...

Reply:
Added

```
 Collecting support bundle ⠙ sonobuoyI0215 06:46:55.370095   50799 sonobuoy_results.go:132] Sonobuoy collector runing command: kubectl exec -n sonobuoy sonobuoy -c kube-sonobuoy -- /sonobuoy splat /tmp/sonobuoy
 Collecting support bundle ⠼ sonobuoyI0215 06:46:56.623800   50799 sonobuoy_results.go:99] Sonobuoy collector found file: 202402151442_sonobuoy_ca07019f-7be9-4c54-9449-e1611c7cc5a8.tar.gz
```